py-rattler/rattler/package/package_name.py

```diff
@@ -71,6 +71,15 @@ def normalized(self) -> str:
         """
         return self._name.normalized
 
+    def __eq__(self, other: object) -> bool:
+        """
+        Returns True if this instance represents the same PackageName as `other`.
```

Comment:
Maybe add a doctest so we dont break this in the future? :)

Reply:
A comparison with a string would also be nice to have! E.g. 

```python
PackageName("foo") == "foo"
```

Not required for this PR